src/extensions/CanvasWidget/CanvasWidget.tsx

```diff
@@ -89,25 +89,36 @@ const CanvasWidget = (props: Props) => {
 
     // To learn about callback refs: https://reactjs.org/docs/hooks-faq.html#how-can-i-measure-a-dom-node
     const [divElement, setDivElement] = useState<HTMLDivElement | null>(null)
+    const [hasFocus, setHasFocus] = useState<boolean>(false)
     const divRef = React.useCallback((elmt: HTMLDivElement) => {
         // this should get called only once after the div has been written to the DOM
         // we set this div element so that it can be used below when we set the canvas
         // elements to the layers
         setDivElement(elmt)
     }, [])
 
+    useEffect(() => {
+        if (divElement) {
+            (divElement as any)['_hasFocus'] = hasFocus
+        }
+    }, [hasFocus, divElement])
+
     useEffect(() => {
         // this should be called only when the divElement has been first set (above)
         // or when the layers (prop) has changed (or if preventDefaultWheel has changed)
         // we set the canvas elements on the layers and schedule repaints
         if (!divElement) return
+        if (!layers) return
+        const stopScrollEvent = (e: Event) => {
+            if ((divElement as any)['_hasFocus']) {
```

Comment:
Here is where we check the divElement for the _hasFocus field. As stated above we do not have access to the state in this listener callback.